controllers/kubeadmconfig_controller.go

```diff
@@ -300,6 +302,68 @@ func (r *KubeadmConfigReconciler) Reconcile(req ctrl.Request) (ctrl.Result, erro
 	return ctrl.Result{}, nil
 }
 
+// reconcileDiscovery ensure that joinConfiguration.Discovery is properly set for the joining node.
+// The implementation func respect user provided discovery configurations, but in case some of them are missing, a valid BootstrapToken object
+// is automatically injected into joinConfiguration.Discovery.
+// This allows to simplify configuration UX, by providing the option to delegate to CAPBK the configuration of kubeadm join discovery.
+func (r *KubeadmConfigReconciler) reconcileDiscovery(cluster *capiv1alpha2.Cluster, discovery *kubeadmv1beta1.Discovery, log logr.Logger) error {
```

Comment:
KubeadmConfigReconciler already has a `logr.Logger`. We don't need to pass one in here. But we should create a new one at the top of this function if some logging scope is needed.